src/plumbing/graph.cljc

```diff
@@ -125,28 +125,45 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Compiling and running graphs
 
+#?(:clj (declare interpreted-eager-compile))
 #?(:clj
 (defn eager-compile
   "Compile graph specification g to a corresponding fnk that is optimized for
    speed. Wherever possible, fnks are called positionally, to reduce the
    overhead of creating and destructuring maps, and the return value is a
    record, which is much faster to create and access than a map.  Compilation
    is relatively slow, however, due to internal calls to 'eval'."
-  [g]
-  (if (fn? g)
-    g
-    (let [g (for [[k sub-g] (->graph g)]
-              [k (eager-compile sub-g)])]
-      (graph-positional/positional-flat-compile (->graph g))))))
+  ([g] (eager-compile g {}))
+  ([g {:keys [positional-limit]
+       :or {positional-limit graph-positional/max-graph-size}}]
+   (let [eager-compile (fn eager-compile [g]
+                         (when (some? g)
+                           (if (fn? g)
+                             g
+                             (let [g* (for [[k sub-g] (->graph g)]
+                                        (do (assert sub-g)
```

Comment:
(same question)

Reply:
I took a closer look and realized I could remove it.